packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp

```diff
@@ -1176,13 +1270,28 @@ void vCycle(const int l, ///< ID of current level
       }
       else if (coarseSolverType == "amg") // use AMG as coarse level solver
       {
+        const bool coarseSolverRebalance = coarseSolverData->get<bool>("coarse solver rebalance");
 
         // Extract the hierarchy from the coarseSolverData
         RCP<Hierarchy> amgHierarchy = coarseSolverData->get<RCP<Hierarchy>>("amg hierarchy object");
 
         // Run a single V-cycle
+        if(coarseSolverRebalance==false){
+          amgHierarchy->Iterate(*compRhs, *compX, 1, true);
 
-        amgHierarchy->Iterate(*compRhs, *compX, true, 1);
+        } else {
+          RCP<const Import> rebalanceImporter = coarseSolverData->get<RCP<const Import> >("rebalanceImporter");
+
+          RCP<Vector> rebalancedRhs = VectorFactory::Build(rebalanceImporter->getTargetMap());
+          RCP<Vector> rebalancedX = VectorFactory::Build(rebalanceImporter->getTargetMap());
+          rebalancedRhs->doImport(*compRhs, *rebalanceImporter, Xpetra::INSERT);
+          rebalancedX->doImport(*compX, *rebalanceImporter, Xpetra::INSERT);
+
+          amgHierarchy->Iterate(*rebalancedRhs, *rebalancedX, 1, true);
+
+          compRhs->doExport(*rebalancedRhs, *rebalanceImporter, Xpetra::INSERT);
+          compX->doExport(*rebalancedX, *rebalanceImporter, Xpetra::INSERT);
```

Comment:
@lucbv How do I want to do this? Is the export OK or do I want to build the correct `import` and do a `doImport`

Reply:
In my mind the above is not correct, I have asked Chris about it, let us see what he says...

Reply:
Hum, looking carefully at the `doExport` documentation in Tpetra I see that with an importer as input it performs a reverse import operation so that above actually makes more sense to me know!

Reply:
Does `rebalancedX` actually need to be communicated? Or are its values just ignored? Calling `Iterate` with last arg `true` seems to suggest that you don't need this.